### PR TITLE
Add a peek function for the class MapDataset

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**

This pull request adds a peek function aiming to plot the counts map (summed over axis) overlayed with the masks, the excess map (under the same representation) and the exposure. This PR is associated to the Issue #3499 .

**Dear reviewer**
If other plot is prefered, do not hesitate. If you wish a different plot of the mask, do not hesitate also.

### CODE REVIEW

**Comment by adonath** [comment]
For reference, this is what the current plot looks like:
```python3
from gammapy.datasets import MapDataset

dataset = MapDataset.read("$GAMMAPY_DATA/cta-1dc-gc/cta-1dc-gc.fits.gz", name="cta_dataset")
dataset.peek()
```
image

**Comment by adonath** [comment]
Here is a possible way to illustrate the masks on top the actual data:

```python3
import matplotlib.pyplot as plt
from gammapy.datasets import MapDataset
from gammapy.maps import RegionGeom
import numpy as np


dataset = MapDataset.read("$GAMMAPY_DATA/cta-1dc-gc/cta-1dc-gc.fits.gz", name="cta_dataset")


geom = RegionGeom.create("galactic;circle(0, 0, 2)")
dataset.mask_fit = dataset.counts.geom.region_mask([geom.region])

counts = dataset.counts.sum_over_axes()
_, ax, _ = counts.smooth("0.1 deg").plot()

ax.set_title("Counts")
mask_fit = dataset.mask_fit.reduce_over_axes(func=np.logical_or)
mask_fit.plot_mask(ax=ax, label="Mask safe", alpha=0.2)

mask_safe = dataset.mask_safe.reduce_over_axes(func=np.logical_or)
mask_safe.plot_mask(ax=ax, label="Mask safe", hatches=["///"], colors="w")
```
image

**Comment by adonath** [comment]
In general there is much more information to show. Just a few suggestions:
- Except for residuals I would always  choose the `afmhot` colormap
- Maybe smooth the counts a bit
- Show background?
- Show current residuals?
- Show predicted number of counts from sources?
- Show masks on top of the data as I have illustrated above?
- Show PSF info? Such as containment radius at image center?
- Show EDisp info such as kernel at the image center?

**Comment by adonath** [comment]
Currently there is now spectral information shown for the `MapDataset`. This might be fine, because users could always do:
```python3
image = dataset.to_image()
image.peek()

spec = dataset.to_spectrum_dataset(region=None) # this will show the integrated spectrum for the whole image
spec.peek()
```

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -1725,6 +1732,131 @@ def to_image(self, name=None):
...
+        fig : `~matplotlib.figure.Figure`
+            Figure to add AxesSubplot on.
+
+        Returns
+        -------
+        ax1, ax2, ax3 : `~matplotlib.axes.AxesSubplot`
+            Counts, excess and exposure.
+        """
+        fig = get_figure(fig, 16, 4)
+        ax1, ax2, ax3 = fig.subplots(1, 3)
```
Comment (adonath) [inline gammapy/datasets/map.py:1845]: Creating the subplots this way misses the WCS info for the plot. You have to pass the `subplot_kw`, see e.g. https://github.com/gammapy/gammapy/blob/master/gammapy/maps/core.py#L1028

**Review by AtreyeeS — COMMENTED** [review]
Thanks for starting this @bkhelifi !

Adding an npred() map does seem quite useful. I do not know if the psf and and edisp kernel at the center is very illuminating to standard users. I agree with @adonath that we can merge this now and add additional plots later.

I personally prefer the `peek` to be a set of quick and simple plots (peek takes no parameters).

File: gammapy/datasets/map.py [hunk 2]
```diff
@@ -1725,6 +1732,131 @@ def to_image(self, name=None):
...
+
+        if self.mask_fit:
+            self.mask_fit_image.plot(ax=ax, **kwargs_fit)
+
+        if self.mask_safe:
+            self.mask_safe_image.plot(ax=ax, **kwargs_safe)
+
+        return ax
+
+    def plot_excess(self, ax=None, kwargs_excess=None, **kwargs):
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/map.py:1802]: I agree this is maybe too fine grained API....

**Comment by adonath** [comment]
@bkhelifi Any chance to get this in by Wednesday?

**Comment by registerrier** [comment]
I have modified the PR such that: 
- there are no extra plot methods
- counts, excess, background and exposure are plotted
- so far no smoothing is applied (is there a good default?)
- plot mask use haches and transparency
- info on the IRFs might be needed, but the better approach is likely to add a `peek` method on `PSFMap` and `EDispMap`

**Review by adonath — APPROVED** [review]
Thanks @bkhelifi  and @registerrier! No further comments from my side..

**Pull request merged**